File: leetCode/python/p58_LengthOfLastWord/lc58.py
#!/usr/bin/python3

class Solution:
    '''
        Given a string s consisting of words and spaces, 
        return the length of the last word in the string.

        A word is a maximal substring consisting of non-space characters only.


    '''
    def lastWord(self, s: str) -> str:
        
        last_word = ''
        length = len(s)

        for i in range(length-1, 0, -1):
            
            if s[i] == ' ' and i != 0:
                return last_word[::-1]
            else:
                last_word += s[i]

    def lengthOfLastWord(self, s: str) -> int:
        
        last_word = self.lastWord(s.strip())
        
        return len(last_word) 
        
if __name__ == '__main__':

    s = Solution()

    # Test 1:
    
    print('Test 1:')
    
    str1 = 'Hello World'
    
    print('Text: \"' + str1, end = '')
    print('\"') 

    print('Length of the last word: ' + str(s.lengthOfLastWord(str1)))

    # Test 2:
    
    print('\nTest 2:')
    
    str2 = '   fly me   to   the moon  '
    
    print('Text: \"' + str2, end = '')
    print('\"') 

    print('Length of the last word: ' + str(s.lengthOfLastWord(str2)))

    # Test 3:

    print('\nTest 3:')
    
    str3 = 'luffy is still joyboy'
    
    print('Text: \"' + str3, end = '')
    print('\"') 

    print('Length of the last word: ' + str(s.lengthOfLastWord(str3)))  

File: leetCode/cpp/p58_LengthOfLastWord/lc58.cpp
#include <iostream>
#include <string>

using std::cout;
using std::endl;

class Solution {

public:
    static int lengthOfLastWord(std::string s);
};

int Solution::lengthOfLastWord(std::string s) {
    
    int len = 0;

    for(int i = 0; i < s.length(); i++) {
    
        if(i != 0 && s[i] != ' ') {
            if(s[i-1] != ' ') len++;
            else len = 1;
        }
        else continue;
    }

    return len;
}

int main(void) {
    
    // Test 1:
    std::string s1 = "Hello World";
    cout << "Input string 1: " << '\"' << s1 << '\"' << endl;
    cout << "Length of last word: " << Solution::lengthOfLastWord(s1) << endl;
    
    // Test 2:
    std::string s2 = "   fly me   to   the moon  ";
    cout << "Input string 2: " << '\"' << s2 << '\"' << endl;
    cout << "Length of last word: " << Solution::lengthOfLastWord(s2) << endl;

    // Test 3:
    std::string s3 = "luffy is still joyboy";
    cout << "Input string 3: " << '\"' << s3 << '\"' << endl;
    cout << "Length of last word: " << Solution::lengthOfLastWord(s3) << endl;
            
    return EXIT_SUCCESS; 
}
